hazelcast/src/test/java/com/hazelcast/map/AbstractMapQueryIterableTest.java

```diff
@@ -0,0 +1,262 @@
+    }
+
+    @Test
+    public void test_Next_Returns_Value_On_NonEmptyMap() {
+        String value = randomString();
+
+        // random partition id for testing
+        fillMap(mapProxy, randomPartitionId(), 1, value);
+
+        final Iterator<String> iterator = getIterable(10,
+                new GetValueProjection<String>(), Predicates.alwaysTrue()).iterator();
+        final String next = iterator.next();
+        assertEquals(value, next);
+    }
+
+    @Test
+    public void test_Next_Returns_Value_On_NonEmptyMap_and_HasNext_Returns_False_when_Item_Consumed() {
+        String value = randomString();
+        fillMap(mapProxy, randomPartitionId(), 1, value);
+
+        final Iterator<String> iterator = getIterable(10,
+                new GetValueProjection<String>(), Predicates.alwaysTrue()).iterator();
+        final String next = iterator.next();
+        assertEquals(value, next);
+        boolean hasNext = iterator.hasNext();
+        assertFalse(hasNext);
+    }
+
+    @Test
+    public void test_HasNext_Returns_True_On_NonEmptyMap() {
+        fillMap(mapProxy, randomPartitionId(), 1, randomString());
+
+        final Iterator<String> iterator = getIterable(10,
+                new TestProjection(), Predicates.alwaysTrue()).iterator();
+        assertTrue(iterator.hasNext());
+    }
+
+
+    @Test
+    public void test_with_projection_and_true_predicate() {
+        fillMap(mapProxy, randomPartitionId(), 100, randomString());
+        final Iterator<String> iterator = getIterable(10,
+                new TestProjection(), Predicates.alwaysTrue()).iterator();
+        final ArrayList<String> projected = collectAll(iterator);
+
+        final Collection<String> actualValues = mapProxy.values();
+        assertEquals(actualValues.size(), projected.size());
+
+        for (String value : actualValues) {
+            assertTrue(projected.contains("dummy" + value));
+        }
+    }
+
+    @Test
+    public void test_iterable_withMultiplePartitions_shouldHaveEqualSizeWithValues() {
+        fillMap(mapProxy, randomPartitionId(), 100, randomString());
+        fillMap(mapProxy, randomPartitionId(), 100, randomString());
+        fillMap(mapProxy, randomPartitionId(), 100, randomString());
+        fillMap(mapProxy, randomPartitionId(), 1, randomString());
+        fillMap(mapProxy, randomPartitionId(), 1, randomString());
+
+        final Iterator<String> iterator = getIterable(10,
+                new TestProjection(), Predicates.alwaysTrue()).iterator();
+        final ArrayList<String> projected = collectAll(iterator);
+
+        final Collection<String> actualValues = mapProxy.values();
+        assertEquals(actualValues.size(), projected.size());
+
+        for (String value : actualValues) {
+            assertTrue(projected.contains("dummy" + value));
+        }
+    }
+
+    @Test
+    public void test_with_projection_and_predicate() {
+        final MapProxyImpl<String, Integer> intMap =
```

Comment:
We are not using the `mapProxy` here, hence not testing the client map proxy version